.github/workflows/typecheck.yml

```diff
@@ -0,0 +1,29 @@
+name: Typecheck
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+      - development
+
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
```

Comment:
why not v3.12?

Reply:
I want to test with annotations that are compatible with the oldest Python version we support. There's a similar setting in `pyproject.toml` for pyright itself!

Reply:
Shouldn't we be setting this to Python 3.7, looking at https://github.com/icesat2py/icepyx/blob/1cf650486c6220b355bc1f0350f41e4167be4a61/pyproject.toml#L7

Reply:
Probably yeah for 1.x, and then update everything to reflect a new version policy in 2.x? Related discussion here: https://github.com/icesat2py/icepyx/issues/569

Reply:
Cool, I'm just asking because I saw you using `Union[..., ...]` in places, and was wondering if we could use `... | ...` instead, but that's only available with Python 3.10 - https://docs.python.org/3.10/whatsnew/3.10.html#pep-604-new-type-union-operator

Reply:
Yeah, I was originally doing that and that caused our 3.9 unit tests to fail :laughing:  So in either case, `3.10` is not correct here. We can run Ruff's pyupgrade ruleset to automatically migrate these :)

Reply:
I think it may not make sense to set to 3.7; I can't install from `requirements.txt` and `requirements-dev.txt`  in Python 3.7 using conda or pip, and can only install in Python 3.8 with pip, not conda. 3.9 installs just fine. Setting to 3.9!

dfe4c09

Reply:
Should we change this to 3.9 too?

Reply:
Do you mean the `requires-python` setting in `pyproject.toml`? I don't have a strong feeling on it, so I erred on the side of not touching it to minimize the scope of the PR. Happy to do it whatever way makes sense to everyone else!

Reply:
> I think we should also change from `~=` compatibility pinning to `>=3.9`.

Yes, this needs to go into release v1.3.0 !!

Reply:
In that case, opened a separate PR so we can be sure that happens :) https://github.com/icesat2py/icepyx/pull/608